### DISCUSSION TITLE
Filesystem & Path does not use themselves

### DISCUSSION THREAD

**Discussion by donaldinou** [symfony/symfony#57768] [Q&A] [answered]
Hello,

Reading code, I've notice that `Symfony\Component\Filesystem\Filesystem` and `Symfony\Component\Filesystem\Path` could have methods in common (`isAbsolutePath`, `makeRelative`, `makeAbsolute` ...), but each implements own logic.

Is there a reason of that? What's the point? 

Any "merge" soon? For example:
```php
// ...
class Filesystem
{
    // ...
    
    public function makePathRelative(string $endPath, string $startPath): string
    {
        return Path::makeRelative($endPath, $startPath);
    }
    
    // ...
}
```

**Comment by donaldinou** [comment]
Hi,

I can't stay with unanswer question :sob: :smile: 
Am I in the right place or did I make a mistake and so can anyone told me where to go?

Thanks

**Comment by stof** [comment]
the main reason is probably that `Path` has been introduced a lot later than `Filesystem`, and nobody has performed the check for common code to clean this.

**Comment by donaldinou** [comment]
@stof Is this something relevant to propose work on it and submit PR or not?

**Comment by nicolas-grekas** [comment, marked as answer]
https://github.com/symfony/symfony/pull/62283
